vsts/build_repo.ps1

```diff
@@ -12,4 +12,38 @@ else
     Write-Host "Pull request build detected"
 }
 
-mvn -DRUN_PROVISIONING_TESTS="$Env:runProvisioningTests" -DRUN_DIGITAL_TESTS="$Env:runDigitalTwinTests" -DRUN_IOTHUB_TESTS="$Env:runIotHubTests" -DIS_PULL_REQUEST="$isPullRequestBuild" install -T 2C
\ No newline at end of file
+# This repo can only run unit tests when using Java 8 currently due to outdated unit test code packages. Because of that,
+# we can only run e2e tests for Java versions other than Java 8.
+if (!($env:JAVA_VERSION).equals("8"))
+{
+    #Build the repo with Java 8 as configured in the project itself
+    mvn install -DskipTests -T 2C
+
+    # go into the e2e tests folder so the next mvn install only runs the e2e tests instead of running e2e tests + unit tests.
+    cd "./iot-e2e-tests"
+}
+
+# Set the Java home equal to the Java version under test. Currently, Azure Devops hosted agents only support Java 8, 11 and 17 currently
```

Comment:
extremely small nit: can be reworded to remove overuse of "currently".

Reply:
```suggestion
# Set the Java home equal to the Java version under test. Currently, Azure Devops hosted agents only support Java 8, 11 and 17
```